src/insights_client/tests/test_metrics.py

```diff
@@ -285,3 +285,115 @@ def test_metrics_post_event_proxy(post, config_file_factory, rhsm_config_file_fa
         proxies={"https": "http://user:password@localhost:3128"},
     )
 
+@mark.parametrize(("config", "delenv", "setenv", "sysargv", "is_offline"), (
+    ("", ("INSIGHTS_OFFLINE"), (), ["insights-client"], False),
+    ("offline=True\n", ("INSIGHTS_OFFLINE"), (), ["insights-client"], True),
+    ("", ("INSIGHTS_OFFLINE"), (), ["insights-client", "--offline"], True),
+    ("", ("INSIGHTS_OFFLINE"), (("INSIGHTS_OFFLINE", "True"),), ["insights-client"], True),
+))
+def test_is_offline(monkeypatch, config, delenv, setenv, sysargv, is_offline, config_file_factory):
+    '''
+    Verify that _is_offline() produces correct output from the given configurations
+    '''
+    # offline not specified
+    for d in delenv:
+        monkeypatch.delenv(d, raising=False)
+    for k, v in setenv:
+        monkeypatch.setenv(k, v)
+    monkeypatch.setattr("sys.argv", sysargv)
+    config_file = config_file_factory(config)
+    cfg = configparser.RawConfigParser()
+    cfg.read(config_file.name)
+    assert _is_offline(cfg) is is_offline
+
+@patch("insights_client.metrics._is_offline", Mock(return_value=True))
+@patch("insights_client.metrics._proxy_settings")
+def test_offline_no_init(_proxy_settings, config_file_factory, rhsm_config_file_factory):
+    '''
+    Verify that when the metrics client is set to offline, no further initialization is done
+    '''
+    config_file = config_file_factory("")
+    rhsm_config_file = rhsm_config_file_factory()
+    m = MetricsHTTPClient(config_file=config_file.name, rhsm_config_file=rhsm_config_file.name)
+    assert m.offline
+    assert not hasattr(m, 'base_url')
+    assert not hasattr(m, 'port')
+    assert not m.cert
+    assert m.verify
```

Comment:
If we want to test this is the unchanged default value not overwritten by _rhsm_rhsm_repo_ca_cert_.

```suggestion
    assert m.verify is True
```